packages/aws-cdk/lib/init-templates/app/csharp/add-project.hook.ts

```diff
@@ -6,28 +6,40 @@ export const invoke: InvokeHook = async (targetDirectory: string) => {
   const slnPath = path.join(targetDirectory, "src", "%name.PascalCased%.sln");
   const csprojPath = path.join(targetDirectory, "src", "%name.PascalCased%", "%name.PascalCased%.csproj");
 
-  const child = child_process.spawn('dotnet', [ 'sln', slnPath, 'add', csprojPath ], {
-    // Need this for Windows where we want .cmd and .bat to be found as well.
-    shell: true,
-    stdio: [ 'ignore', 'pipe', 'inherit' ]
-  });
+  try {
+    await runCommand('dotnet', 'sln', slnPath, 'add', csprojPath);
+  } catch (err) {
+    throw new Error(`Could not add project %name.PascalCased%.csproj to solution %name.PascalCased%.sln: ${err}`);
+  }
+
+  try {
+    await runCommand('dotnet', 'add', csprojPath, 'package', 'Amazon.Jsii.Analyzers');
+  } catch (err) {
+    throw new Error(`Could not add Amazon.Jsii.Analyzers to project %name.PascalCased%.csproj: ${err}`);
+  }
+};
 
-  await new Promise<string>((resolve, reject) => {
-    const stdout = new Array<any>();
+function runCommand(command: string, ...args: string[]): Promise<string> {
```

Comment:
Let's move this to `lib/` so we don't have to duplicate it so much.

Reply:
But... those files are supposed to be stand-alone... They're part of the template but won't be included in the generated bundle... If I add stuff to `lib/` I suddenly have them not self-contained anymore, and the whole templating becomes a lot more complicated...?